src/Components/Admin/ObjectEditor/object.js

```diff
@@ -0,0 +1,263 @@
+import React, { useEffect, useState } from "react";
+import { makeStyles } from "@material-ui/core/styles";
+import { useErrorHandler } from "react-error-boundary";
+import Select from "react-select";
+import AddIcon from "@material-ui/icons/Add";
+// import DialogTitle from "@material-ui/core/DialogTitle";
+// import DialogContent from "@material-ui/core/DialogContent";
+// import DialogActions from "@material-ui/core/DialogActions";
+// import Dialog from "@material-ui/core/Dialog";
+import { Button, IconButton } from "@material-ui/core";
+import {
+    DeleteForever,
+    KeyboardArrowDown,
+    KeyboardArrowUp,
+} from "@material-ui/icons";
+
+// import { getPuzzle } from "../../../lib/puzzleEndpoints";
+import { getPuzzle, editPuzzle } from "../../../lib/puzzleEndpoints";
+
+const useStyles = makeStyles((theme) => ({
+    page: {
+        marginTop: theme.spacing(8),
+    },
+    sceneAndTransitionContainer: {
+        minWidth: 600,
+    },
+    introContainer: {
+        minWidth: 850,
+        display: "flex",
+    },
+    container: {
+        paddingTop: theme.spacing(12),
+    },
+    dragAndDropContainer: {
+        display: "flex",
+        alignItems: "center",
+        marginLeft: "16px",
+    },
+    emptyButtonsContainer: {
+        display: "flex",
+        width: "100%",
+        flexDirection: "column",
+        flexGrow: 1,
+        justifyContent: "center",
+        alignItems: "space-around",
+    },
+    buttonContainer: {
+        display: "flex",
+        width: "100%",
+        justifyContent: "center",
+        marginTop: theme.spacing(12),
+        marginBottom: theme.spacing(4),
+    },
+    button: {
+        display: "flex",
+        justifyContent: "center",
+        padding: "16px 12px",
+        backgroundColor: "#E2E5ED",
+        borderRadius: "12px",
+    },
+}));
+
+export default function ObjectEditor({
+    match: {
+        params: { sceneId, objectId },
+    },
+}) {
+    const classes = useStyles();
+    const handleError = useErrorHandler();
+    const [puzzleBody, setPuzzleBody] = useState({});
+    const [puzzleType, setPuzzleType] = useState("");
+    const [animationsJson, setAnimationsJson] = useState({});
+    const [isInteractable, setIsInteractable] = useState(false);
+    const [texts, setTexts] = React.useState([]);
+
+    const puzzleTypeList = [
+        { value: "text-pane", label: "text-pane" },
```

Comment:
Can we update the actual text that is shown to the user to be more human-readable? `text-pane` doesn't sound very friendly lol, same for stuff like `rotation-controls`